utils/jwtUtil.js

```diff
@@ -19,10 +19,52 @@ const User = require('../db/index').User;
 const Token = require('../db/index').Token;
 const Collector = require('../db/index').Collector;
 const Bot = require('../db/index').Bot;
+const Profile = require('../db/index').Profile;
 const Promise = require('bluebird');
 const jwtVerifyAsync = Promise.promisify(jwt.verify);
 
 /**
+ * Adds a key and its value to the passed in object.
+ * @param  {Object} object - The object to which the key value pair needs to be
+ * assgined
+ * @param  {String} key - The key that will be added to the object
+ * @param  {String} value  - The value that will be assigned to the key
+ */
+function assignKeyValue(object, key, value) {
+  object[key] = value;
+} // assignKeyValue
+
+/**
+ * Assigns user related information to the request header.
+ * @param  {Object}  req  - The request object
+ * @param  {String}  tokenName - Token name parsed from the token
+ * @param  {Boolean} isAdmin  - Flag to indicate if the request was made by an
+ * Admin user
+ */
+function assignUserHeaders(req, tokenName, isAdmin) {
+  assignKeyValue(req.headers, 'UserName', req.user.name);
+  assignKeyValue(req.headers, 'TokenName', tokenName);
+  assignKeyValue(req.headers, 'ProfileName', req.user.profile.name);
+  assignKeyValue(req.headers, 'IsAdmin', isAdmin);
+  assignKeyValue(req.headers, 'IsCollector', false);
+  assignKeyValue(req.headers, 'IsBot', false);
+} // assignUserHeaders
+
+/**
+ * Assigns collector related information to the request header
+ * @param  {Object} req - The request object
+ * @param  {String} userName  - The username parsed from the token
+ * @param  {String} tokenName - The tokenname parsed from the token
+ */
+function assignCollectorHeaders(req, userName, tokenName) {
+  assignKeyValue(req.headers, 'UserName', userName);
+  assignKeyValue(req.headers, 'TokenName', tokenName);
+  assignKeyValue(req.headers, 'IsCollector', true);
+  assignKeyValue(req.headers, 'IsAdmin', false);
+  assignKeyValue(req.headers, 'IsBot', false);
+} // assignCollectorHeaders
```

Comment:
Would assigning 'ProfileName' to empty string actually set a header with an empty string, or does it delete the named header? In either case, we should do that.